### PR TITLE
Fix 3D-IWGAN

### PR DESCRIPTION
**Pull request by Jean-Francois-Lafleche**
### Description
- Fix code style errors.
- Fix Gradient Penalty calculation code to allow for stable training.
- Add weight initialization to match original implementation (https://github.com/EdwardSmith1884/3D-IWGAN).

### Tests
OS: Ubuntu 18.04
GPU: RTX 2080Ti
Driver: 435.21
CUDA: 10.1

Commands:
- `python train.py --modelnet-root /data/modelnet --cache-dir cache/` - trained for 1660 epochs
- `python train.py --modelnet-root /data/modelnet --cache-dir cache/ --resume` 
- `python eval.py` - sample output below (1660 epochs is not fully trained):
3d-iwgan

### CODE REVIEW

File: examples/GANs/3D-IWGAN/eval.py [hunk 1]
```diff
@@ -12,51 +12,41 @@
...
-Commandline arguments
-"""
 parser = argparse.ArgumentParser()
-parser.add_argument('-expid', type=str, default='GAN', help='Unique experiment identifier.')
-parser.add_argument('-device', type=str, default='cuda', help='Device to use')
-parser.add_argument('-batchsize', type=int, default=50, help='Batch size.')
-
+parser.add_argument('--logdir', type=str, default='log', help='Directory to log data to.')
+parser.add_argument('--expid', type=str, default='GAN', help='Unique experiment identifier.')
+parser.add_argument('--device', type=str, default='cuda', help='Device to use')
```
Comment (Caenorst) [inline examples/GANs/3D-IWGAN/eval.py]: '.' missing at the end of the `help` field

File: examples/GANs/3D-IWGAN/train.py [hunk 2]
```diff
@@ -14,55 +14,41 @@
...
-Dataset
-"""
-train_set = kal.datasets.ModelNet(root ='../../datasets/',categories = args.categories, \
-			single_view= True, download = True)
-dataloader_train = DataLoader(train_set, batch_size=args.batchsize, shuffle=True, 
-	num_workers=8)
+parser = argparse.ArgumentParser()
+parser.add_argument('--modelnet-root', type=str, help='Root directory of the ModelNet dataset.')
+parser.add_argument('--cache-dir', type=str, default=None, help='Path to write intermediate representation to.')
+parser.add_argument('--expid', type=str, default='GAN', help='Unique experiment identifier.')
```
Comment (Caenorst) [inline examples/GANs/3D-IWGAN/train.py]: Is this really  "Unique" if it have the same default id than in https://github.com/NVIDIAGameWorks/kaolin/blob/master/examples/GANs/3DGAN/train.py#L32 ?

File: examples/GANs/3D-IWGAN/utils.py [hunk 3]
```diff
@@ -13,68 +13,26 @@
...
+def calculate_gradient_penalty(D, real_samples, fake_samples):
+    """Calculates the gradient penalty loss"""
+    # Random weight for interpolation between real and fake samples
+    eta = torch.rand((real_samples.size(0), 1, 1, 1), device=real_samples.device)
+    # Get random interpolation between real and fake samples
+    interpolates = (eta * real_samples + ((1 - eta) * fake_samples)).requires_grad_(True)
+    # calculate probability of interpolated examples
+    d_interpolates = D(interpolates)
+    # Get gradient w.r.t. interpolates
+    fake = torch.ones_like(d_interpolates).requires_grad_(False).to(real_samples.device)
```
Comment (Caenorst) [inline examples/GANs/3D-IWGAN/utils.py]: Use parameters `device` and `requires_grad` of `torch.ones_like` instead of doing it after initialising

**Review by Caenorst — COMMENTED** [review]
Have you tried the `resume` option that you added ?

File: examples/GANs/3D-IWGAN/eval.py (hunk 1 above)
Comment (Jean-Francois-Lafleche) [reply examples/GANs/3D-IWGAN/eval.py]: Nice catch, fixed.

File: examples/GANs/3D-IWGAN/train.py (hunk 2 above)
Comment (Jean-Francois-Lafleche) [reply examples/GANs/3D-IWGAN/train.py]: Fixed.

File: examples/GANs/3D-IWGAN/utils.py (hunk 3 above)
Comment (Jean-Francois-Lafleche) [reply examples/GANs/3D-IWGAN/utils.py]: Ah, I missed that! Good catch. Fixed.

**Comment by Jean-Francois-Lafleche** [comment]
--resume flag has been tested. Updated the PR description accordingly.

**Review by Caenorst — APPROVED** [review]
LGTM, thanks for the good work !
